cmd/gitserver/internal/git/gitcli/blame_test.go

```diff
@@ -430,6 +430,24 @@ func TestBlameHunkReader(t *testing.T) {
 	})
 }
 
+func BenchmarkBlameBytes(b *testing.B) {
+	for range b.N {
+		rc := io.NopCloser(strings.NewReader(testGitBlameOutputIncremental2))
+		reader := newBlameHunkReader(rc)
+		defer reader.Close()
+
+		for {
+			_, err := reader.Read()
+			if errors.Is(err, io.EOF) {
+				break
+			} else if err != nil {
+				b.Fatalf("blameHunkReader.Read failed: %s", err)
+			}
+		}
```

Comment:
`defer` in a for loop can actually be pretty expensive, and seems to be skewing the results a bit. Would recommend closing manually. I suspect you'll get an even better improvement 🙂  

```suggestion

		for {
			_, err := reader.Read()
			if errors.Is(err, io.EOF) {
				break
			} else if err != nil {
				b.Fatalf("blameHunkReader.Read failed: %s", err)
			}
		}
		reader.Close()
```